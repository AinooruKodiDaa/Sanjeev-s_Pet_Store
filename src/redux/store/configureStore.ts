import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
// import { composeWithDevTools } from 'redux-devtools-extension';
import rootSaga from '../sagas'; // Import your root saga
import usersReducer from '../reducers/usersReducer';
import rootReducer from '../reducers';
// // Combine all reducers into a root reducer
// const rootReducer = combineReducers({
//   users: usersReducer,
//   doctors: doctorsReducer,
//   appointments: appointmentsReducer,
// });

// Create saga middleware
const sagaMiddleware = createSagaMiddleware();

// Create the Redux store with middleware and enhancers
const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

// Run the root saga
sagaMiddleware.run(rootSaga);

export default store;
