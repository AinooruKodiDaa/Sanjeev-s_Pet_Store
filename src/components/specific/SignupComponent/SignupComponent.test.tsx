import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { SignupComponent } from './';
import MockRouter from '../../../__mocks/mockRouter';


const mockStore = configureStore([]);

describe('SignupComponent', () => {
  let wrapper: any; // Change the type to any if necessary
  let store: any;

  beforeEach(() => {
    store = mockStore({
      // Provide initial state if needed
    });

    wrapper = mount(
      <Provider store={store}>
        <MockRouter>
          <SignupComponent />
        </MockRouter>
      </Provider>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
