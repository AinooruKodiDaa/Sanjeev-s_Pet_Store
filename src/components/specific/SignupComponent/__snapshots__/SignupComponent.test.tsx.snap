// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupComponent should match snapshot 1`] = `ReactWrapper {}`;
